app/services/sorting/Calculator.java

```diff
@@ -0,0 +1,99 @@
+package services.sorting;
+
+import services.dataAccess.proto.PostProto.Post;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Calculator {
+
+    private static final Double LIKE_WEIGHT = 0.9;
+    private static final Double COMMENT_WEIGHT = 0.5;
+    private static final Double SHARE_WEIGHT = 1.1;
+    private static final Long POST_EXPIRY = 259200000L; // three days (in milliseconds)
+
+    /**
+     * Calculates and inserts popularity score to each post in a list of posts
+     *
+     * @param posts list of posts
+     * @return the same list of posts, but with each post now containing a popularity score
+     */
+    public List<Post> calculatePopularityScoreOfAllPosts(List<Post> posts) {
+        List<Post> calculatedPosts = new ArrayList<>();
+
+        posts.forEach(post -> {
```

Comment:
whole method could be replaced with `return posts.stream().map(p -> calculatePopularityAndRebuild(post)).collect(Collectors.toList())` to be even simpler